packages/block-library/src/pattern/edit.js

```diff
@@ -0,0 +1,56 @@
+/**
+ * WordPress dependencies
+ */
+import { useSelect, useDispatch } from '@wordpress/data';
+import { useEffect } from '@wordpress/element';
+import {
+	store as blockEditorStore,
+	InnerBlocks,
+} from '@wordpress/block-editor';
+
+const PatternEdit = ( { attributes, clientId, isSelected } ) => {
+	const selectedPattern = useSelect( ( select ) =>
+		select( blockEditorStore ).__experimentalGetParsedPattern(
+			attributes.slug
+		)
+	);
+	const isInnerBlockSelected = useSelect( ( select ) => {
+		const { getBlock, hasSelectedInnerBlock } = select( blockEditorStore );
+
+		const block = getBlock( clientId );
+		const hasInnerBlocks = !! ( block && block.innerBlocks.length );
+		return hasInnerBlocks && hasSelectedInnerBlock( clientId, true );
+	} );
```

Comment:
1. I think we could rename this to something like `hasSelection` and absorb everything in one place: selection of the pattern block and selection of any of its children.
2. Totally subjective, but IMO we can let the selectors deal with empty states, i.e. we don't need to check the existence of the block itself or its children.

```suggestion
	const hasSelection = useSelect(
		( select ) =>
			isSelected ||
			select( blockEditorStore ).hasSelectedInnerBlock( clientId, true ),

		[ isSelected, clientId ]
	);
```

Reply:
Nice